protected/humhub/modules/search/controllers/MentioningController.php

```diff
@@ -33,37 +46,157 @@ public function getAccessRules()
         ];
     }
 
+    /**
+     * Find all users and spaces on mentioning request from RichText editor
+     *
+     * @return \yii\web\Response
+     */
     public function actionIndex()
     {
-        Yii::$app->response->format = 'json';
+        $keyword = (string)Yii::$app->request->get('keyword');
+
+        // Find users
+        $users = User::find()
+            ->visible()
+            ->search($keyword)
+            ->limit($this->module->mentioningSearchBoxResultLimit)
+            ->orderBy(['user.last_login' => SORT_DESC])
+            ->all();
 
         $results = [];
+        foreach ($users as $user) {
+            $results[] = $this->getUserResult($user);
+        }
+
+        $results = $this->appendMentioningSpaceResults($keyword, $results);
+
+        return $this->asJson($results);
+    }
+
+    /**
+     * Find space members on mentioning request from RichText editor on Post form
+     *
+     * @return \yii\web\Response
+     * @throws HttpException
+     */
+    public function actionSpace()
+    {
+        $spaceId = (int)Yii::$app->request->get('id');
         $keyword = (string)Yii::$app->request->get('keyword');
 
-        // Add user results
-        $query = User::find()->visible()->search($keyword);
-        foreach ($query->limit(10)->all() as $container) {
-            $results[] = [
-                'guid' => $container->guid,
-                'type' => 'u',
-                'name' => $container->getDisplayName(),
-                'image' => UserImage::widget(['user' => $container, 'width' => 20]),
-                'link' => $container->getUrl()
-            ];
-        };
-
-        // Add space results
-        $query = Space::find()->visible()->search($keyword);
-        foreach ($query->limit(10)->all() as $container) {
-            $results[] = [
-                'guid' => $container->guid,
-                'type' => 's',
-                'name' => $container->getDisplayName(),
-                'image' => SpaceImage::widget(['space' => $container, 'width' => 20]),
-                'link' => $container->getUrl()
-            ];
+        $space = Space::findOne(['id' => $spaceId]);
+        if (!$space || !$space->can(CreatePost::class)) {
+            throw new HttpException(403, 'Access denied!');
```

Comment:
Use of `Yii::t('SearchModule.error', 'Access denied!')` as it would not be translatable and issues would be created to make it translatable later, that is of course if @luke- doesn't see this to be necessary?

Reply:
@ArchBlood Thank you, but in my opinion it is not necessary. Actually, 'throw HttpException(403)` should also be sufficient. Especially with such internal errors that should properly be caught via the frontend.